src/ctapipe/utils/astro.py

```diff
@@ -4,63 +4,230 @@
 
     Returns
     -------
-    Astropy table:
-       List of all stars after cuts with names, catalog numbers, magnitudes,
-       and coordinates
+    astropy.table.Table
+        List of all stars after applying the cuts, with the same keys as the input table `stars`.
     """
-    from ctapipe.utils import get_table_dataset
-
-    catalog = get_table_dataset("yale_bright_star_catalog5", role="bright star catalog")
-
-    starpositions = SkyCoord(
-        ra=Angle(catalog["RAJ2000"], unit=u.deg),
-        dec=Angle(catalog["DEJ2000"], unit=u.deg),
-        frame="icrs",
-        copy=False,
-    )
-    catalog["ra_dec"] = starpositions
-
-    if magnitude_cut is not None:
-        catalog = catalog[catalog["Vmag"] < magnitude_cut]
+    stars_ = deepcopy(stars)
+    if magnitude_cut:
+        try:
+            stars_ = stars_[stars_[band] < magnitude_cut]
+        except KeyError:
+            raise ValueError(
+                f"The requested catalogue has no compatible magnitude for the {band} band"
+            )
 
     if radius is not None:
-        if pointing is None:
+        if pointing:
+            stars_["separation"] = stars_["ra_dec"].separation(pointing)
+            stars_ = stars_[stars_["separation"] < radius]
+        else:
             raise ValueError(
                 "Sky pointing, pointing=SkyCoord(), must be "
                 "provided if radius is given."
             )
-        separations = catalog["ra_dec"].separation(pointing)
-        catalog["separation"] = separations
-        catalog = catalog[separations < radius]
 
-    catalog.remove_columns(["RAJ2000", "DEJ2000"])
+    return stars_
+
+
+def get_star_catalog(
+    catalog: str | StarCatalog, magnitude_cutoff: float = 8.0, row_limit: int = 1000000
+) -> Table:
+    """
+    Utility function to download a star catalog for the get_bright_stars function.
+
+    Parameters
+    ----------
+    catalog : str or ctapipe.utils.astro.StarCatalog
+        Name of the catalog to be used. Usable names are found in the Enum StarCatalog. Default: Yale.
+    magnitude_cutoff : float, optional
+        Maximum value for magnitude used in lookup. Default is 8.0.
+    row_limit : int, optional
+        Maximum number of rows for the star catalog lookup. Default is 1000000.
+
+    Returns
+    -------
+    astropy.table.Table
+        List of all stars after cuts with catalog numbers, magnitudes,
+        and coordinates as SkyCoord objects including proper motion.
+    """
+    from astroquery.vizier import Vizier
+
+    if isinstance(catalog, str):
+        catalog = StarCatalog[catalog]
+    catalog_dict = catalog.value
+
+    vizier = Vizier(
+        catalog=catalog_dict.directory,
+        columns=catalog_dict.columns,
+        row_limit=row_limit,
+    )
+
+    stars = vizier.query_constraints(Vmag=f"<{magnitude_cutoff}")[0]
+    meta = catalog_dict._asdict()
+    meta["magnitude_cutoff"] = magnitude_cutoff
+
+    stars.meta["Catalog"] = meta
```

Comment:
How are theses nested metadata items handled with the FITS export?

Reply:
Ah, in an astropy custom way in `COMMENT` cards:
```
COMMENT --BEGIN-ASTROPY-SERIALIZED-COLUMNS--                                    
COMMENT datatype:                                                               
COMMENT - name: RAICRS                                                          
COMMENT   unit: deg                                                             
COMMENT   datatype: float64                                                     
COMMENT   format: '{:12.8f}'                                                    
COMMENT   description: '? alpha, degrees (ICRS, Epoch=J1991.25) (H8)'           
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: POS_EQ_RA_MAIN}                                               
COMMENT   - {width: 12}                                                         
COMMENT   - {precision: '8'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: DEICRS                                                          
COMMENT   unit: deg                                                             
COMMENT   datatype: float64                                                     
COMMENT   format: '{:12.8f}'                                                    
COMMENT   description: '? delta, degrees (ICRS, Epoch=J1991.25) (H9)'           
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: POS_EQ_DEC_MAIN}                                              
COMMENT   - {width: 12}                                                         
COMMENT   - {precision: '8'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: pmRA                                                            
COMMENT   unit: mas / yr                                                        
COMMENT   datatype: float64                                                     
COMMENT   format: '{:8.2f}'                                                     
COMMENT   description: '*? Proper motion mu_alpha.cos(delta), ICRS(H12)'        
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: POS_EQ_PMRA}                                                  
COMMENT   - {width: 8}                                                          
COMMENT   - {precision: '2'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: pmDE                                                            
COMMENT   unit: mas / yr                                                        
COMMENT   datatype: float64                                                     
COMMENT   format: '{:8.2f}'                                                     
COMMENT   description: '*? Proper motion mu_delta, ICRS (H13)'                  
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: POS_EQ_PMDEC}                                                 
COMMENT   - {width: 8}                                                          
COMMENT   - {precision: '2'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: Vmag                                                            
COMMENT   unit: mag                                                             
COMMENT   datatype: float32                                                     
COMMENT   format: '{:5.2f}'                                                     
COMMENT   description: '? Magnitude in Johnson V (H5)'                          
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: PHOT_JHN_V}                                                   
COMMENT   - {width: 5}                                                          
COMMENT   - {precision: '2'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: BTmag                                                           
COMMENT   unit: mag                                                             
COMMENT   datatype: float32                                                     
COMMENT   format: '{:6.3f}'                                                     
COMMENT   description: '? Mean BT magnitude (H32)'                              
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: PHOT_TYCHO_B}                                                 
COMMENT   - {width: 6}                                                          
COMMENT   - {precision: '3'}                                                    
COMMENT   - values: {'null': .nan}                                              
COMMENT - name: HIP                                                             
COMMENT   datatype: int32                                                       
COMMENT   description: Identifier (HIP number) (H1)                             
COMMENT   meta: !!omap                                                          
COMMENT   - {ucd: ID_MAIN}                                                      
COMMENT   - {width: 6}                                                          
COMMENT meta:                                                                   
COMMENT   __serialized_columns__: {}                                            
COMMENT --END-ASTROPY-SERIALIZED-COLUMNS-- 
```

I'd prefer if you store the single metadata fields we need to save using a flat hierarchy that can be directly stored to the FITS header.

Reply:
Sorry, I didn't check before whether it is getting in properly.

What if we use json dumps? The result will look like this:

```

> stars = Table.read('./hipparcos.fits.gz')
> stars.meta

OrderedDict([('ID', 'I_239_hip_main'),
             ('NAME', 'I/239/hip_main'),
             ('description', 'The Hipparcos Main Catalogue'),
             ('CATALOG',
              '{"directory": "I/239/hip_main", "coordinates": {"frame": "icrs", "epoch": "J1991.25", "RA": {"column": "RAICRS", "unit": "deg"}, "DE": {"column": "DEICRS", "unit": "deg"}}, "columns": ["RAICRS", "DEICRS", "pmRA", "pmDE", "Vmag", "BTmag", "HIP"], "record": "hipparcos_star_catalog", "magnitude_cutoff": 8.0}')])
```

Reply:
I would prefer setting proper FITS headers instead of dumping a json string into a header card that will certainly require using lots of `CONTINUE` cards.

For some of these, there are even reserved appropriate keywords:

* `JEPOCH = <float>` for the epoch
* `RADECSYS = "ICRS"` for the reference frame

Reply:
Now it looks like this:
```
ORIGIN  = 'CTAPIPE '                                                            
JEPOCH  =              1991.25                                                  
RADESYS = 'ICRS    '                                                            
MAGCUT  =                  8.0                                                  
BAND    = 'V       '                                                            
CATALOG = 'Hipparcos'                                                           
REFERENC= 'I/239/hip_main'                                                      
COLUMNS = 'RAICRS_DEICRS_pmRA_pmDE_Vmag_BTmag_HIP' 
```

Reply:
Great!